app/views/observations/show/_collection_numbers.html.erb

```diff
@@ -27,12 +27,14 @@
                                  id: number.id,
                                  params: { back: observation.id }
                                )) %> |
-           <%= link_with_query(:REMOVE.t,
-                               collection_number_remove_observation_path(
-                                 collection_number_id: number.id,
-                                 params: { observation_id: observation.id }
-                               ),
-                               { data: { confirm: :are_you_sure.t } }) %>]
+           <%= patch_button(name: :REMOVE.t,
+                            path: collection_number_remove_observation_path(
+                              collection_number_id: number.id,
+                              observation_id: observation.id,
+                              q: get_query_param,
+                            ),
+                            data: { confirm: :are_you_sure.t }) %>]
```

Comment:
Is this the best way to solve the problem?
Should the `q` param be included?

Reply:
Yes! To the first one. 
re: q, i confess i don't know when to keep it. But there doesn't seem to be any harm in passing it.

Thanks for catching this, it's a bewildering number of new CRUD routes I'm making.